### PR TITLE
New pull_request_template

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
+- [ ] I have run through the testing script to make sure current functionality is unchanged
+- [ ] I have done relevant tests that prove my fix is effective or that my feature works
+- [ ] I have spent time testing out edge cases for my feature
+- [ ] I have updated the test script or pull request template if necessary
+- [ ] New and existing unit tests pass locally with my changes
 
-* 
+## Reviewer template (please paste into your review):
+- [ ] I have made sure I understand which parts of the codebase could be affected by this change.
+- [ ] I have proposed additional tests to prevent side effects, if appropriate.
```
Comment (tfrcarvalho) [inline pull_request_template.md]: This part is not necessary, IMO

File: pull_request_template.md [hunk 2]
```diff
@@ -1,35 +1,42 @@
...
-# Test and Reproduce
-Outline the steps to test or reproduce the PR here.
+## Reproduce:
+Outline the steps required to see the intended effect of your PR:
+
+## Related code:
+- Which code (which methods/actions/components) could this change interact with? Please trace dependencies for every method/action/component that was modified:
 
-1. 
+- Who would be the best person to review each one of these code parts?
```
Comment (tfrcarvalho) [inline pull_request_template.md]: This should evaluated by the reviewer, using common sense.
You should only approve if you feel you have the knowledge to do it.

Comment (cdebourcy) [reply pull_request_template.md]: Agreed. But author can also put some thought into who to request specifically. While most people have some knowledge about most parts of the code, it could be useful to think about who knows the most about its ins and outs, previous gotchas, reasons for specific optimizations, etc

**Comment by cdebourcy** [comment]
Agree this is all just a documentation & reminder of common sense, but reminders are sometimes useful :)
I'm onboard with adding more unit testing as well.
